SantaFinder.ts

```diff
@@ -1,110 +1,110 @@
+        })        
+        let count = availableNodes[0].possiblePresentees.length
+        availableNodes = availableNodes.filter((node) => {
+            return node.possiblePresentees.length === count
         })
-        console.log(chosenCycle)
-    }
 
-    getRandomSantas() {
-        this.chooseRandomCycle()
+        // Choose random santa and presentee (within priority constraints)
+        const random = Math.floor(Math.random() * availableNodes.length);
+        const node = availableNodes[random]
+        const presentee = presentees.filter((presentee) => { return node.possiblePresentees.includes(presentee[0]) })[0]
+        
+        return {santa: node.id, presentee: presentee[0]}
+    }
 
+    getSantas() {
         const santas: { [key: string]: { name: string, presenteeName: string } } = {}
-        this.getSantaIds().forEach(santaId => {
-            const presenteeId = this.getSantaById(santaId).chosenPresenteeId
-            santas[santaId] = {
-                name: this.getSantaById(presenteeId).name,
-                presenteeName: this.getSantaById(santaId).name
+        this.santas.forEach(santa => {
+            const presenteeId = santa.chosenPresenteeId
+            santas[santa.id] = {
+                name: santa.name,
+                presenteeName: this.getSantaById(presenteeId).name
             }
         })
-        return santas
-    }
-
-    private findNextSanta(nodeId: string, edges: Array<string>, path: Array<string>) {
-        const currentPath = [...path]
-        currentPath.push(nodeId)
-
-        if (currentPath.length === this.getSantaIds().length) {
-            // Does endNode match firstNode
-            if (this.santas[nodeId].possiblePresentees.includes(currentPath[0])) {
-                this.possibleCycles.push(currentPath)
-            }
-            return
-        }
 
-        edges.forEach(nextNodeId => {
-            if (!currentPath.includes(nextNodeId)) {
-                this.findNextSanta(nextNodeId, this.santas[nextNodeId].possiblePresentees, currentPath)
-            }
-        })
+        return santas
     }
 
     private getPossiblePresenteesById(santaId: string) {
-        return this.getSantaIds().filter(presenteeId => {
+        return this.santas.filter(presentee => {
             const currentSanta = this.getSantaById(santaId)
-            const isPresenteeExcluded = currentSanta.excludedPresentees.includes(presenteeId)
-            const isOtherPerson = santaId != presenteeId
+            const isPresenteeExcluded = currentSanta.excludedPresentees.includes(presentee.id)
+            const isOtherPerson = santaId != presentee.id
             return !isPresenteeExcluded && isOtherPerson
-        })
-    }
-
-    getSantaIds() {
-        return Object.keys(this.santas)
+        }).map((santa) => santa.id)
     }
 
     getSantaById(id: string) {
-        return this.santas[id]
+        return this.santas.filter((santa) => { return santa.id === id })[0]
```

Comment:
```suggestion
        return this.santas.find((santa) => santa.id === id )
```

find() statt filter()[0] und shortForm :)